horreum-backend/src/main/java/io/hyperfoil/tools/horreum/action/SlackPluginBase.java

```diff
@@ -0,0 +1,77 @@
+package io.hyperfoil.tools.horreum.action;
+
+import java.util.concurrent.TimeUnit;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.inject.Inject;
+
+import org.jboss.logging.Logger;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.RequestOptions;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpResponse;
+
+public abstract class SlackPluginBase {
+   protected static final Logger log = Logger.getLogger(SlackPluginBase.class);
+
+   @Inject
+   Vertx vertx;
+
+   @Inject
+   SlackClient client;
+
+   @Inject
+   Instance<BodyFormatter> formatters;
+
+   protected void requireProperties(JsonNode configuration, String... properties) {
+      for (String property : properties) {
+         if (!configuration.hasNonNull(property)) {
+            throw missing(property);
+         }
+      }
+   }
+
+   private IllegalArgumentException missing(String property) {
+      return new IllegalArgumentException("Configuration is missing property '" + property + "'");
+   }
+
+   protected Uni<HttpResponse<Buffer>> post(String path, JsonNode secrets, JsonNode payload) {
+      // Token should NOT be in the dataset!
+      String token = secrets.path("token").asText();
+      if (token == null || token.isBlank()) {
+         throw new IllegalArgumentException("Missing access token!");
+      }
+      log.infof("POST %s (%s): %s", path, token, payload.toString());
+      RequestOptions options = new RequestOptions()
+            .setHost("slack.com")
+            .setPort(443)
+            .setURI("/api/chat.postMessage")
+            .setSsl(true);
```

Comment:
These values should be configurable (I see it is not in the github example, but I think that is an oversight), with these specific values as the default. This will allow us to make changes to service config if either a) anything changes in the future b) if we want to mock a slack service in testing

Please see https://github.com/Hyperfoil/Horreum/blob/master/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/action/HttpAction.java#L36 for configuration examples with a default values

We could then define difference profiles in `application.properties`;

```
%prod.horreum.action.slack.host=slack.com
%prod.horreum.action.slack.port=443
%prod.horreum.action.slack.ssl=true

%test.horreum.action.slack.host=localhost
%test.horreum.action.slack.port=8081
%test.horreum.action.slack.ssl=false

```

Reply:
Sounds like a good idea